examples/webgl_materials_cars.html

```diff
@@ -66,541 +61,334 @@
+				carParts.glass.forEach( function ( part ) { part.material = glassMat; } );
 
 			}
 
 			function onWindowResize() {
 
-				windowHalfX = window.innerWidth / 2;
-				windowHalfY = window.innerHeight / 2;
-
 				camera.aspect = window.innerWidth / window.innerHeight;
 				camera.updateProjectionMatrix();
 
 				renderer.setSize( window.innerWidth, window.innerHeight );
 
 			}
 
-			function initCarButton( car ) {
-
-				$( car ).addEventListener( 'click', function() {
+			function onDriveModeToggle() {
 
-					if ( ! CARS[ car ].object ) {
+				car.enabled = !car.enabled;
+				controls.enabled = !car.enabled;
 
-						loader.load( CARS[ car ].url, function( geometry ) { createScene( geometry, car ) } );
-
-					} else {
-
-						switchCar( car );
-
-					}
-
-				}, false );
+				controls.reset();
+				carModel.position.copy( origin );
 
 			}
 
-			function $( id ) { return document.getElementById( id ) }
-			function button_name( car, index ) { return "m_" + car  + "_" + index }
-
-			function switchCar( car ) {
-
-				for ( var c in CARS ) {
+			function update() {
 
-					if ( c != car && CARS[ c ].object ) {
+				var delta = clock.getDelta();
 
-						CARS[ c ].object.visible = false;
-						CARS[ c ].buttons.style.display = "none";
+				if ( carModel && !controls.enabled ) {
 
-					}
-				}
-
-				CARS[ car ].object.visible = true;
-				CARS[ car ].buttons.style.display = "block";
+					car.update( delta );
 
-				$( "car_name" ).innerHTML = CARS[ car ].name + " model";
-				$( "car_author" ).innerHTML = CARS[ car ].author;
-
-			}
+					updateCamera( delta );
 
-			function createButtons( materials, car ) {
+					resetPosition();
 
-				var buttons, i, src = "";
+					// keep the light (and shadow) pointing in the same direction as the car rotates
+					lightHolder.rotation.y = -carModel.rotation.y;
 
-				for( i = 0; i < materials.length; i ++ ) {
+				} else {
 
-					src += '<button id="' + button_name( car, i ) + '">' + materials[ i ][ 0 ] + '</button> ';
+					controls.update();
 
 				}
 
-				buttons = document.createElement( "div" );
-				buttons.innerHTML = src;
-
-				$( "buttons_materials" ).appendChild( buttons );
-
-				return buttons;
+				stats.update();
 
 			}
 
-			function attachButtonMaterials( materials, faceMaterials, material_indices, car ) {
-
-				for( var i = 0; i < materials.length; i ++ ) {
+			function updateCamera( delta ) {
 
-					$( button_name( car, i ) ).counter = i;
-					$( button_name( car, i ) ).addEventListener( 'click', function() {
+				carModel.getWorldPosition( cameraTarget );
+				cameraTarget.y = 2.5;
+				cameraTarget.z += distance;
 
-						for ( var j = 0; j < material_indices.length; j ++ ) {
+				camera.position.lerp( cameraTarget, delta * damping );
+				camera.lookAt( carModel.position );
 
-							faceMaterials[ material_indices [ j ] ] = materials[ this.counter ][ 1 ];
+      }
 
-						}
-
-					}, false );
-
-				}
-
-			}
+      function resetPosition() {
```

Comment:
Um, this looks still not correct. BTW: You can easily detect wrong formatting by enabling invisible characters in your editor.

Reply:
Thanks, should all be fixed now.